### ISSUE TITLE
fix makisu dry run

### ISSUE THREAD

**Issue by plyr4** [go-vela/community#246]
## Description

Removing the `pushes:` tag (as stated in the documentation) should disabled registry pushes, and execute a "dry run" docker build.

In its current state, removing `pushes:` will still publish the built image if you provide `replicas`.

Adding `pushes: []` will cause a failed build via an attempted incomplete request to push the image to an empty registry host.

> ERROR	failed to push image: failed to push image: check manifest exists for image app/davidvader/heyvela:latest: check manifest exists: network error: failed to fallback from https to http, original https error: Head "https:///v2/app/davidvader/heyvela/manifests/latest": http: no Host in request URL,
243 |  

## Solution

Add a note in the documentation for removing `replicas` when attempting a dry run build.

## YAML to Reproduce

```yaml
steps:
  - name: docker_build
    image: target/vela-makisu:v0.1.2
    parameters:
      registry: docker.index.io
      repo: docker.index.io/davidvader/heyvela
      tag: docker.index.io/davidvader/heyvela
      replicas:
        - docker.index.io/davidvader/heyvela:v1
      context: .
      file: Dockerfile
    secrets: [ docker_username, docker_password ]
```

**Comment by davidvader** [comment]
At first this looked like a bug with the interpretation of `pushes` but rather a user error from providing `replicas` without any `pushes`.
Not necessarily a bug, but functionality/user-experience around this could be improved.
At the very least, we could add a note to the documentation for using this plugin.

**Comment by colindean** [comment]
I encountered this yesterday, but slightly differently: I thought that a default value of `pushes` — omitting it — enabled the `tag` to push, and that if I wanted Makisu _not_ to push, I'd need to specify `pushes: []`. So, I thought for a while that my `latest` had been pushing! Turns out not, and thus some incredible confusion when an error from a removed code path kept appearing like a specter from the past.

I would prefer an option that disables _all_ pushing, something suitable for a dry run or build-only step such that the _next_ step can do the push and only do so on master/main+push events while the build step always builds for all push/pull_request events.

However, I did catch in my build step that `replicas` pushes so I didn't have that in place. But my confusion around why my `replicas` were pushing but not `latest` was a doozy until I realized I needed `pushes` to push.
